graphdatascience/procedure_surface/arrow/catalog/relationship_arrow_endpoints.py

```diff
@@ -0,0 +1,191 @@
+        )
+
+    def stream(
+        self,
+        G: Graph,
+        relationship_types: Optional[List[str]] = None,
+        relationship_properties: Optional[list[str]] = None,
+        *,
+        concurrency: Optional[Any] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+    ) -> DataFrame:
+        config_input = {
+            "graph_name": G.name(),
+            "relationship_types": relationship_types or ["*"],
+            "concurrency": concurrency,
+            "sudo": sudo,
+            "log_progress": log_progress,
+            "username": username,
+        }
+
+        endpoint = "v2/graph.relationships.stream"
+        if relationship_properties and relationship_properties != []:
+            config_input["relationship_properties"] = relationship_properties
+            endpoint = "v2/graph.relationshipProperties.stream"
+
+        config = ConfigConverter.convert_to_gds_config(**config_input)
+
+        job_id = JobClient.run_job(self._arrow_client, endpoint, config)
+        result = JobClient.stream_results(self._arrow_client, G.name(), job_id)
+
+        return result
+
+    def write(
+        self,
+        G: Graph,
+        relationship_type: str,
+        relationship_properties: Optional[list[str]] = None,
+        *,
+        concurrency: Optional[Any] = None,
+        write_concurrency: Optional[Any] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        job_id: Optional[Any] = None,
+    ) -> RelationshipsWriteResult:
+        if self._write_back_client is None:
+            raise ValueError("Write back is only available if a database connection is provided.")
+
+        config_input = {
+            "graph_name": G.name(),
+            "relationship_types": [relationship_type],
+            "concurrency": concurrency,
+            "sudo": sudo,
+            "log_progress": log_progress,
+            "username": username,
+            "job_id": job_id,
+        }
+
+        endpoint = "v2/graph.relationships.stream"
+        if relationship_properties and relationship_properties != []:
+            config_input["relationship_properties"] = relationship_properties
+            endpoint = "v2/graph.relationshipProperties.stream"
+
+        config = ConfigConverter.convert_to_gds_config(**config_input)
+
+        job_id = JobClient.run_job(self._arrow_client, endpoint, config)
+
+        write_result = self._write_back_client.write(
+            G.name(),
+            job_id,
+            concurrency=write_concurrency if write_concurrency is not None else concurrency,
+            relationship_type_overwrite=relationship_type,
+        )
+
+        written_relationships = (
+            write_result.written_relationships if hasattr(write_result, "written_relationships") else 0
```

Comment:
when would it not have `written_relationships`?

also, is it not `relationships_written`?

Reply:
no, it is `written_relationships`, because this is remote ops, not plugin